Add Hugo info to README

lucperkins opened the pull request:
To make sure that people are on the same page when attempting to run the site locally, I added a brief section to the README that provides instructions and makes it clear which version of Hugo they'll need to install.

kbhawkey commented on README.md:
```diff
@@ -13,14 +15,31 @@ For more information about contributing to the Kubernetes documentation, see:
+
+The Kubernetes documentation is built using the [Hugo](https://gohugo.io) static site generator. See the [official Hugo documentation](https://gohugo.io/getting-started/installing/) for Hugo installation instructions.
+
+> The site currently requires Hugo version **0.46**.
```
As I recall, I installed 0.46 first and the build still failed. I then installed the extended version, checked the Hugo docs which I thought indicated that the `resources` function is available only in the extension. I can re-check if needed.

kbhawkey replied:
Here is what I found, locally:
- a version previous to 0.46 (ex. 0.42.1) emits a stack trace of errors about the resources method not found
- version 0.46 emits an info message that the resources method is not available in this version
- version 0.46 with extensions works w/o error

lucperkins replied:
@kbhawkey That sounds consistent with what I would expect. I think the current recommendation should be fine, though I'll add a note about the `INFO`-level log in case users who have the non-extended version are confused.

kbhawkey replied:
@lucperkins, I am unable to build with Hugo 0.46. The extended version works, though. Are the two generated files (.content, .json), from a successful build, necessary?

Here is the output using 0.46:
```$ hugo server
Building sites … ERROR 2018/08/16 10:38:23 error: failed to transform resource: TOCSS: failed to transform "sass/styles.sass" (text/x-sass): this feature is not available in your current Hugo version
Total in 19371 ms
Error: Error building site: logged 1 error(s)```

GuessWhoSamFoo replied:
Based on outcome of this, `/docs/contribute/intermediate/` will need an update for the section about running locally.

mistyhacks commented on README.md:
```diff
@@ -13,17 +15,34 @@ For more information about contributing to the Kubernetes documentation, see:
+
+The Kubernetes documentation is built using the [Hugo](https://gohugo.io) static site generator. See the [official Hugo documentation](https://gohugo.io/getting-started/installing/) for Hugo installation instructions.
+
+> The site currently requires Hugo version **0.46**.
```
I think you can set this as a Hugo variable in the and use that variable here. It stinks that we have to have it also in the README. Any suggestions, @bep ?

bep replied:
This is a README.md meant to be rendered and read on GitHub. No variables available. But I would make the language less specific to make it simpler to maintain.

mistyhacks commented on README.md:
````diff
@@ -13,17 +15,34 @@ For more information about contributing to the Kubernetes documentation, see:
+hugo server --ignoreCache --disableFastRender
+```
+
+This will start the local Hugo server on port 1313. Open up your browser to http://localhost:1313 to view the site. As you make changes to docs' source files, Hugo will update immediately and force a browser refresh.
````
s/docs//

mistyhacks replied:
s/will update immediately/immediately updates

mistyhacks commented on README.md:
```diff
@@ -13,17 +15,34 @@ For more information about contributing to the Kubernetes documentation, see:
+
 ## Building the site using Docker
 
 If you'd like, you can build the Kubernetes docs using Docker. To get started, build the image locally:
```
Not your change, but s/If you'd like, you/You

mistyhacks commented on README.md:
````diff
@@ -13,17 +15,34 @@ For more information about contributing to the Kubernetes documentation, see:
 If you'd like, you can build the Kubernetes docs using Docker. To get started, build the image locally:
 
 ```bash
 make docker-image
````
Does this mean that the `make` command is actually doing the underlying command? The way this is presented might lead people to think they need to run both commands.

lucperkins replied:
Good call. I've removed the underlying command and kept only the `make` command. Users can investigate the underlying commands on their own.

lucperkins commented:
@bep @mistyhacks I've changed the approach a bit. Now I link to the `netlify.toml` file to provide the current Hugo version, which seems more maintainable to me.

@mistyhacks I've also addressed the wording concerns that you raised.

lucperkins commented on README.md:
```diff
@@ -13,41 +15,36 @@ For more information about contributing to the Kubernetes documentation, see:
 ## Building the site using Docker
 
-If you'd like, you can build the Kubernetes docs using Docker. To get started, build the image locally:
+You can build the Kubernetes docs using [Docker](https://docker.com). To get started, make sure that you have Docker running and build the image locally:
```
Yes, this currently works fine

lucperkins commented on README.md:
````diff
@@ -13,41 +15,36 @@ For more information about contributing to the Kubernetes documentation, see:
 ```
 
-As when building without using a Docker container, the results of the build will be published to the `public` directory (the default output directory for [Hugo](https://gohugo.io), the static site generator used to build this site).
+As when building without using a Docker container, the results of the build will be published to the `public` directory (the default output directory for Hugo).
````
Done

mistyhacks commented:
/approve
